File: mini-commerce/src/main/java/me/hero/minicommerce/item/adapter/in/web/dto/CreateItemRequest.java
package me.hero.minicommerce.item.adapter.in.web.dto;

import lombok.Data;
import me.hero.minicommerce.item.application.port.in.dto.CreateItemDto;

@Data
public class CreateItemRequest {
  private String name;
  private Long price;

  public CreateItemRequest() {}

  public CreateItemDto toDto() {
    return new CreateItemDto(null, name, price);
  }

}

File: mini-commerce/src/main/kotlin/me/hero/minicommerce/item/adapter/in/web/dto/CreateItemRequest.kt
package me.hero.minicommerce.item.adapter.`in`.web.dto

import me.hero.minicommerce.item.application.port.`in`.dto.CreateItemDto


data class CreateItemRequest (
    val name: String,
    val price: Long,

) {
    fun toDto(): CreateItemDto {
        return CreateItemDto(
            name = name,
            price = price,
        )
    }
}